src/Sylius/Behat/Context/Setup/PromotionContext.php

```diff
@@ -93,6 +103,26 @@ public function thereIsPromotion($promotionName)
     }
 
     /**
+     * @Given the store has promotion :promotionName with coupon :couponCode
+     */
+    public function thereIsPromotionWithCoupon($promotionName, $couponCode)
+    {
+        /** @var CouponInterface $coupon */
+        $coupon = $this->couponFactory->createNew();
+        $coupon->setCode($couponCode);
+
+        $promotion = $this->testPromotionFactory
+            ->createForChannel($promotionName, $this->sharedStorage->get('channel'))
+        ;
+        $promotion->addCoupon($coupon);
+        $promotion->setCouponBased(true);
+
+        $this->promotionRepository->add($promotion);
```

Comment:
Actually, now in ``DocumentRepository``on each ``add`` method object manager is flushed automatically. Its moot is it bad or good behaviour (I think it should be changed), but the fact is, we don't have to flush here.